tests/components/group/test_init.py

```diff
@@ -18,307 +17,375 @@
-        grp1 = group.Group.create_group(self.hass, "Je suis Charlie")
-        grp2 = group.Group.create_group(self.hass, "Je suis Charlie")
-
-        assert grp1.entity_id != grp2.entity_id
-
-    def test_expand_entity_ids_expands_nested_groups(self):
-        """Test if entity ids epands to nested groups."""
-        group.Group.create_group(self.hass, "light", ["light.test_1", "light.test_2"])
-        group.Group.create_group(
-            self.hass, "switch", ["switch.test_1", "switch.test_2"]
-        )
-        group.Group.create_group(
-            self.hass, "group_of_groups", ["group.light", "group.switch"]
-        )
-
-        assert [
-            "light.test_1",
-            "light.test_2",
-            "switch.test_1",
-            "switch.test_2",
-        ] == sorted(group.expand_entity_ids(self.hass, ["group.group_of_groups"]))
-
-    def test_set_assumed_state_based_on_tracked(self):
-        """Test assumed state."""
-        self.hass.states.set("light.Bowl", STATE_ON)
-        self.hass.states.set("light.Ceiling", STATE_OFF)
-        test_group = group.Group.create_group(
-            self.hass, "init_group", ["light.Bowl", "light.Ceiling", "sensor.no_exist"]
-        )
-
-        state = self.hass.states.get(test_group.entity_id)
-        assert not state.attributes.get(ATTR_ASSUMED_STATE)
-
-        self.hass.states.set("light.Bowl", STATE_ON, {ATTR_ASSUMED_STATE: True})
-        self.hass.block_till_done()
-
-        state = self.hass.states.get(test_group.entity_id)
-        assert state.attributes.get(ATTR_ASSUMED_STATE)
-
-        self.hass.states.set("light.Bowl", STATE_ON)
-        self.hass.block_till_done()
-
-        state = self.hass.states.get(test_group.entity_id)
-        assert not state.attributes.get(ATTR_ASSUMED_STATE)
-
-    def test_group_updated_after_device_tracker_zone_change(self):
-        """Test group state when device tracker in group changes zone."""
-        self.hass.states.set("device_tracker.Adam", STATE_HOME)
-        self.hass.states.set("device_tracker.Eve", STATE_NOT_HOME)
-        self.hass.block_till_done()
-        group.Group.create_group(
-            self.hass, "peeps", ["device_tracker.Adam", "device_tracker.Eve"]
-        )
-        self.hass.states.set("device_tracker.Adam", "cool_state_not_home")
-        self.hass.block_till_done()
-        assert STATE_NOT_HOME == self.hass.states.get(f"{group.DOMAIN}.peeps").state
+async def test_setup_group_with_mixed_groupable_states(hass):
+    """Try to set up a group with mixed groupable states."""
+
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("device_tracker.Paulus", STATE_HOME)
+
+    assert await async_setup_component(hass, "group", {})
+
+    await group.Group.async_create_group(
+        hass, "person_and_light", ["light.Bowl", "device_tracker.Paulus"]
+    )
+
+    await hass.async_block_till_done()
+
+    assert STATE_ON == hass.states.get(f"{group.DOMAIN}.person_and_light").state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)